certgen/certgen_test.go

```diff
@@ -164,6 +191,9 @@ func TestIssueCertificate(t *testing.T) {
 			if cert.IsCA {
 				t.Errorf("certificate IsCA = %t, want false", cert.IsCA)
 			}
+			if test.precert && !cert.IsPrecertificate() {
```

Comment:
I think you're missing the converse in this test.
ie.  `!test.precert && cert.IsPrecertificate()`

Reply:
So you might need to do the pattern we see for `gotErr, wantErr`
i.e.

```
if gotPre, wantPre := cert.IsPrecertificate(), test.precert; gotPre != wantPre {
  t.Errorf("certificate type wrong: got precert? %t want precert? %t", gotPre, wantPre)
}
```

Reply:
Done.